packages/flagging/src/openfeature/provider.ts

```diff
@@ -5,57 +5,68 @@
 import type { EvaluationContext, JsonValue, Logger, Paradigm, ResolutionDetails } from '@openfeature/core'
 import { StandardResolutionReasons } from '@openfeature/core'
 import type { Provider } from '@openfeature/web-sdk'
+import type { PrecomputeClient } from '../precomputeClient'
 
 export class DatadogProvider implements Provider {
   runsOn: Paradigm = 'client'
   metadata = {
     name: 'datadog',
   } as const
 
+  private precomputeClient: PrecomputeClient | undefined
+
+  constructor(precomputeClient?: PrecomputeClient) {
+    this.precomputeClient = precomputeClient
+  }
+
   resolveBooleanEvaluation(
-    _flagKey: string,
+    flagKey: string,
     defaultValue: boolean,
     _context: EvaluationContext,
     _logger: Logger
   ): ResolutionDetails<boolean> {
     return {
-      value: defaultValue,
+      value: this.precomputeClient?.getBooleanAssignment(flagKey, defaultValue) ?? defaultValue,
       reason: StandardResolutionReasons.DEFAULT,
     }
   }
 
   resolveStringEvaluation(
-    _flagKey: string,
+    flagKey: string,
     defaultValue: string,
     _context: EvaluationContext,
     _logger: Logger
   ): ResolutionDetails<string> {
     return {
-      value: defaultValue,
+      value: this.precomputeClient?.getStringAssignment(flagKey, defaultValue) ?? defaultValue,
       reason: StandardResolutionReasons.DEFAULT,
     }
   }
 
   resolveNumberEvaluation(
-    _flagKey: string,
+    flagKey: string,
     defaultValue: number,
     _context: EvaluationContext,
     _logger: Logger
   ): ResolutionDetails<number> {
     return {
-      value: defaultValue,
+      value: this.precomputeClient?.getNumericAssignment(flagKey, defaultValue) ?? defaultValue,
       reason: StandardResolutionReasons.DEFAULT,
     }
   }
 
   resolveObjectEvaluation<T extends JsonValue>(
-    _flagKey: string,
+    flagKey: string,
     defaultValue: T,
     _context: EvaluationContext,
     _logger: Logger
   ): ResolutionDetails<T> {
+    const value =
+      typeof defaultValue === 'object' && defaultValue !== null && this.precomputeClient
+        ? (this.precomputeClient.getJSONAssignment(flagKey, defaultValue as object) as T)
+        : defaultValue
     return {
-      value: defaultValue,
+      value,
```

Comment:
openfeature wants to return a `JsonValue` but I don't want to inject the type into the precompute client. the mapping is rough.